File: 位运算/136. 只出现一次的数字.cpp
#include <iostream>
#include <vector>

using namespace std;

class SingleNumber {
public:
    int singleNumber(vector<int> nums) {
        int result = 0;
        for (int num : nums) {
            result ^= num;
        }
        return result;
    }
};

int main() {
    SingleNumber singleNumber;
    cout << singleNumber.singleNumber({4, 12, 34, 12, 4}) << endl;
    cout << singleNumber.singleNumber({4, 12, 34, 12, 4, 5, 34}) << endl;
}

File: 位运算/136. 只出现一次的数字.java
public class SingleNumber {
    public static int singleNumber(int[] nums) {
        int result = 0;
        for (int num : nums) {
            result ^= num;
        }
        return result;
    }

    public static void main(String[] args) {
        System.out.println(singleNumber(new int[]{4, 12, 34, 12, 4}));
        System.out.println(singleNumber(new int[]{4, 12, 34, 12, 4, 5, 34}));
    }
}